tfx/components/base/base_executor_test.py

```diff
@@ -37,21 +44,130 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
     pass
 
 
+class _BeamTestExecutor(base_executor.BeamExecutor):
+  """Fake BeamExecutor for testing purpose only."""
+
+  def beam_io_signature(self, input_dict: Dict[Text, List[types.Artifact]],
+                        output_dict: Dict[Text, List[types.Artifact]],
+                        exec_properties: Dict[Text, Any]
+                        ) -> List[Dict[Text, Any]]:
+    input_signature = {_EXAMPLES: io.TextIOBase}
+    output_signature = {_SHUFFLED_EXAMPLES: io.TextIOBase}
+    return input_signature, output_signature
+
+  def read_inputs(self, pipeline: beam.Pipeline,
+                  input_dict: Dict[Text, List[types.Artifact]],
+                  output_dict: Dict[Text, List[types.Artifact]],
+                  exec_properties: Dict[Text, Any]
+                  ) -> Dict[Text, beam.pvalue.PCollection]:
+    input_dir = input_dict[_EXAMPLES].uri
+    beam_inputs = ({_EXAMPLES: (pipeline
+                                | beam.io.textio.ReadFromText(input_dir))})
```

Comment:
Can you make this `beam.io.textio.ReadFromText(input_dir).with_output_types(Text)`? (see comment below regarding typing)

Reply:
Done, I added that `.with_output_types(Text)`